src/test/java/StringManipulationTest.java

```diff
@@ -119,31 +174,42 @@ public void testRestoreString1()
     }
 
     @Test
-    public void testRestoreString2()
-    {
-        fail("Not yet implemented");
-
-    }
-
-    @Test
-    public void testRestoreString3()
+    public void testRestoreString2_ThrowIllegalArgumentException()
     {
-        fail("Not yet implemented");
+        manipulatedstring.setString("art");
+        int [] array;
+        array=new int[]{1,0};
 
+        assertThrows(IllegalArgumentException.class,
+                ()-> {
+                    String restoreString = manipulatedstring.restoreString(array);
+                });
     }
 
     @Test
-    public void testRestoreString4()
+    public void testRestoreString3_ThrowIndexOutOfBoundsException_GreaterThan()
     {
-        fail("Not yet implemented");
+        manipulatedstring.setString("art");
+        int [] array;
+        array=new int[]{1,0,10};
 
+        assertThrows(IndexOutOfBoundsException.class,
+                ()-> {
+                    String restoreString = manipulatedstring.restoreString(array);
+                });
     }
 
     @Test
-    public void testRestoreString5()
+    public void testRestoreString4_ThrowIndexOutOfBoundsException_LessThan()
     {
-        fail("Not yet implemented");
+        manipulatedstring.setString("art");
+        int [] array;
+        array=new int[]{1,0,-10};
 
+        assertThrows(IndexOutOfBoundsException.class,
+                ()-> {
+                    String restoreString = manipulatedstring.restoreString(array);
+                });
     }
```

Comment:
With the exception of testCount(), all of the test cases check if exceptions are thrown. It may be worth adding unique test cases that test your program logic to ensure that your methods operate as you intend them to.